src/Microsoft.AspNetCore.Mvc.Core/Internal/DisableRequestSizeLimitFilter.cs

```diff
@@ -58,6 +56,10 @@ public void OnAuthorization(AuthorizationFilterContext context)
                     _logger.RequestBodySizeLimitDisabled();
                 }
             }
+            else
+            {
+                _logger.NotMostEffectiveFilter(typeof(IRequestSizePolicy).ToString());
+            }
```

Comment:
The control flow here should probably be changed to an early exit for readability